core/include/usFrameworkEvent.h

```diff
@@ -0,0 +1,217 @@
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef USFRAMEWORKEVENT_H
+#define USFRAMEWORKEVENT_H
+
+#include <iostream>
+#include <memory>
+
+#include "usCoreExport.h"
+#include "usSharedData.h"
+
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable:4251)
+#endif
+
+namespace us {
+
+class Bundle;
+class FrameworkEventData;
+
+/**
+ * \ingroup MicroServices
+ *
+ * An event from the Micro Services framework describing a Framework event.
+ * <p>
+ * <code>FrameworkEvent</code> objects are delivered to listeners connected
+ * via BundleContext::AddFrameworkListener() when an event occurs within the
+ * Framework which a user would be interested in. A <code>Type</code> code 
+ * is used to identify the event type for future extendability.
+ *
+ * @see BundleContext#AddFrameworkListener
+ */
+class US_Core_EXPORT FrameworkEvent
+{
+
+  SharedDataPointer<FrameworkEventData> d;
+
+public:
+
+  /**
+   * A type code used to identify the event type for future extendability.
+   */
+  enum class Type : unsigned int {
+
+	/**
+	 * The Framework is about to be started.
+	 * <p>
+	 * The Framework's
+	 * \link BundleActivator::Start(BundleContext*) BundleActivator Start\endlink method
+	 * is about to be called.
+	 */
+    STARTING,
+
+    /**
+     * The Framework has been started.
+     * <p>
+     * The Framework's
+     * \link BundleActivator::Start(BundleContext*) BundleActivator Start\endlink method
+     * has been executed.
+     */
+    STARTED = 1,
+
+	/**
+	 * An error has occurred.
+	 * <p>
+	 * There was an error associated with a bundle.
+	 */
+    ERROR = 2,
+
+	/**
+	 * An informational event has occurred.
+	 * <p>
+	 * There was an informational event associated with a bundle.
+	 */
+    INFO = 32,
+
+	/**
+	 * A warning has occurred.
+	 * <p>
+	 * There was a warning associated with a bundle.
+	 */
+    WARNING = 16,
+
+    /**
+     * The Framework has been stopped.
+     * <p>
+     * The Framework's
+     * \link BundleActivator::Stop(BundleContext*) BundleActivator Stop\endlink method
+     * has been executed.
+     */
+    STOPPED = 64,
+
+    /**
+     * The Framework is about to be stopped.
+     * <p>
+     * The Framework's
+     * \link BundleActivator::Stop(BundleContext*) BundleActivator Stop\endlink method
+     * is about to be called.
+     */
+    STOPPING,

```

Comment:
You're right, it doesn't exist in the OSGi 6 spec. this was a carry over from using the BundleEvent implementation for FrameworkEvent.

If there are no objections; I'll remove ```STOPPING```.